@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -275,13 +276,35 @@ export class TileGeometryCreator {
      *
      * @param tile - The {@link Tile} to process.
      * @param decodedTile - The decodedTile containing the actual tile map data.
+     * @returns Promise resolved when all textures are ready to render.
      */
-    createAllGeometries(tile: Tile, decodedTile: DecodedTile) {
+    createAllGeometries(tile: Tile, decodedTile: DecodedTile): Promise<void> {
```

Comment:
You can (but don't strictly have to), mark this async, I think from a documentation POV, it is nice.

Reply:
I'm not sure what you mean.
a) If you mean I should just add "async" in front of the method without any further changes, that would actually be misleading from the documentation POV. The method is fully synchronous, ESLint would complain about the missing "async" otherwise.
- If you mean rewrite it to make it asynchronous, we could do it by changing at the very end ```return texturesReady;``` to  ```await texturesReady;```. But I don't see any benefit in that. Since this is a public method, better leave the decision to the calling code. Async callers may await on the promise, sync callers can work with the promise.

Awaiting on every single texture promise is not an option either, cause the promises are generated sequentially, that would make the texture loading sequential.

Reply:
@atomicsulfate , I mean option 1.

Technically the function is synchronous, yes, but it returns a Promise, so the result isn't fully known. You can argue that the function returns without any need to wait, but users of this are going to likely anyway `await/.then()` on the result of this function, so they won't know if it was synchronous or not, they will just use the result.

The [docs](https://stackoverflow.com/questions/45594596/async-function-without-await-in-javascript#comment87236078_45594870) on Mozilla.org also mention:

> Async functions can contain zero or more await expressions.

Another benefit to using async is that any error thrown will cause the promise to reject, so users of the API can use `.catch` and react accordingly. Otherwise they need to wrap this function in a try/catch block, see: https://stackoverflow.com/questions/45594596/async-function-without-await-in-javascript#comment87236078_45594870